benchmarks/run.ml

```diff
@@ -239,107 +258,221 @@ let _ =
   let compile_only = !compile_only in
   let nobyteopt = !nobyteopt in
   let full = !full in
-  let effects = !effects in
   let param = !param in
   let interpreters = read_config conf_file in
   let compile = compile param ~comptime:true in
-  let compile_jsoo ?(effects = false) opts =
+  let compile_jsoo ~wasm ?(effects = `None) opts =
     compile
       (Format.sprintf
-         "js_of_ocaml -q --target-env browser --debug mark-runtime-gen %s %s"
+         "%s -q %s%s %s"
+         (if wasm then "wasm_of_ocaml" else "js_of_ocaml")
+         (if wasm then "" else "--target-env browser --debug mark-runtime-gen ")
          opts
-         (if effects then "--enable=effects" else "--disable=effects"))
+         (match effects with
+         | `None -> "--disable=effects"
+         | `Cps -> "--effects=cps"
+         | `Double_translation -> "--effects=double-translation"))
+  in
+  let js = not !nojs in
+  let wasm = !wasm in
+  let effects =
+    match !effects, wasm with
+    | ((`None | `Cps) as e), _ -> e
+    | (`Double_translation as e), false -> e
+    | `Double_translation, true ->
+        raise
+          (Arg.Bad "option `-effects double-translation` incompatible with option `-wasm`")
   in
   Format.eprintf "Compile@.";
   compile "ocamlc" src Spec.ml code Spec.byte;
   compile "ocamlopt" src Spec.ml code Spec.opt;
-  compile_jsoo "" code Spec.byte code Spec.js_of_ocaml;
-  compile_jsoo "--opt=3" code Spec.byte code Spec.js_of_ocaml_o3;
-  compile_jsoo "--enable=use-js-string" code Spec.byte code Spec.js_of_ocaml_js_string;
-  compile_jsoo "--disable inline" code Spec.byte code Spec.js_of_ocaml_inline;
-  compile_jsoo "--disable deadcode" code Spec.byte code Spec.js_of_ocaml_deadcode;
-  compile_jsoo "--disable compact" code Spec.byte code Spec.js_of_ocaml_compact;
-  compile_jsoo "--disable optcall" code Spec.byte code Spec.js_of_ocaml_call;
-  compile_jsoo ~effects:true "" code Spec.byte code Spec.js_of_ocaml_effects;
   compile "ocamlc -unsafe" src Spec.ml code Spec.byte_unsafe;
   compile "ocamlopt" src Spec.ml code Spec.opt_unsafe;
-  compile_jsoo "" code Spec.byte_unsafe code Spec.js_of_ocaml_unsafe;
+  if wasm then compile_jsoo ~wasm "" code Spec.byte code Spec.wasm_of_ocaml;
+  if js
+  then (
+    compile_jsoo ~wasm "" code Spec.byte code Spec.js_of_ocaml;
```

Comment:
This doesn't look correct. if both js and wasm are set, this will use `wasm_of_ocaml` instead of `js_of_ocaml` to compile the file.
I think it would be better to rename `compile_jsoo` with something like
```ocaml
val compile_js : backend:[`Jsoo | `Wasmoo] -> ...
```